consul-lambda-registrator/extension/extension.go

```diff
@@ -0,0 +1,310 @@
+package extension
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/proxy"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/structs"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/trace"
+)
+
+type Config struct {
+	ServiceName         string
+	ServiceNamespace    string        `envconfig:"CONSUL_SERVICE_NAMESPACE"`
+	ServicePartition    string        `envconfig:"CONSUL_SERVICE_PARTITION"`
+	ServiceUpstreams    []string      `envconfig:"CONSUL_SERVICE_UPSTREAMS"`
+	MeshGatewayURI      string        `envconfig:"CONSUL_MESH_GATEWAY_URI" required:"true"`
+	ExtensionDataPrefix string        `envconfig:"CONSUL_EXTENSION_DATA_PREFIX" required:"true"`
+	RefreshFrequency    time.Duration `envconfig:"CONSUL_REFRESH_FREQUENCY" default:"5m"`
+
+	Store  ParamGetter
+	Events EventProcessor
+	Logger hclog.Logger
+}
+
+type ParamGetter interface {
```

Comment:
My thought (since I didn't explain myself 🙂 ) was, because the SSMClient will potentially be used across packages(?), those packages may all like to reference an interface for the SSMClient. So in that case, it would make sense to put the interface in a shared place. And a mock for the SSMClient could also go in a shared place for tests to reuse (like a `testutil` package or etc).

This is totally fine as-is, though, and your comment makes sense to me!